salt/modules/yumpkg.py

```diff
@@ -3219,3 +3220,75 @@ def list_installed_patches():
         salt '*' pkg.list_installed_patches
     '''
     return _get_patches(installed_only=True)
+
+
+@salt.utils.decorators.path.which('yum-complete-transaction')
+def complete_transaction(cleanup_only=False, recursive=False, max_attempts=3):
+    '''
+    .. versionadded:: Fluorine
+
+    Execute ``yum-complete-transaction``, which is provided by the ``yum-utils`` package.
+
+    cleanup_only
+        Specify if the ``--cleanup-only`` option should be supplied.
+
+    recursive
+        Specify if ``yum-complete-transaction`` should be called recursively
+        (it only completes one transaction at a time).
+
+    max_attempts
+        If ``recursive`` is ``True``, the maximum times ``yum-complete-transaction`` should be called.
+
+    .. note::
+
+        Recursive calls will stop once ``No unfinished transactions left.`` is in the returned output.
+
+    .. note::
+
+        ``yum-utils`` will already be installed on the minion if the package
+        was installed from the Fedora / EPEL repositories.
+
+    CLI example:
+
+    .. code-block:: bash
+
+        salt '*' pkg.complete_transaction
+        salt '*' pkg.complete_transaction cleanup_only=True
+        salt '*' pkg.complete_transaction recursive=True max_attempts=5
+    '''
+
+    return _complete_transaction(cleanup_only, recursive, max_attempts, 1, [])
+
+
+def _complete_transaction(cleanup_only, recursive, max_attempts, run_count, cmd_ret_list):
+    '''
+    .. versionadded:: Fluorine
+
+    Called from ``complete_transaction`` to protect the arguments
+    used for tail recursion, ``run_count`` and ``cmd_ret_list``.
+    '''
+
+    cmd = ['yum-complete-transaction']
+    if cleanup_only:
+        cmd.append('--cleanup-only')
+
+    cmd_ret_list.append(__salt__['cmd.run_all'](
+        cmd,
+        output_loglevel='trace',
+        python_shell=False
+    ))
+
+    if (cmd_ret_list[-1]['retcode'] == salt.defaults.exitcodes.EX_OK and
+            recursive and
+            'No unfinished transactions left.' not in cmd_ret_list[-1]['stdout']):
```

Comment:
I think there is some problem with the indent, isn't it?

Reply:
pep8 seems okay with it either way. Are you thinking the "r" in `recursive` should be under the "c" in `cmd_ret_list` and "'No unfinished" be left where it is?

Reply:
I don't see anything overriding in Salt's guide, but pep8 states:
> This PEP takes no explicit position on how (or whether) to further visually distinguish such conditional lines from the nested suite inside the if-statement.

I'm happy to un-indent `recursive`, or simply drop it to a new line and leave `if (` and indent below that ("Vertical is better").

Reply:
@leeclemens yeah, if that works, fine. Maybe github shows it that way strange, so I just thought that Linter didn't captured that. :wink: 

As of just sharing experiences, I personally would format "operator first", once I use parenthesis. That said, I would do:

```python
if (cmd_ret_list[-1]['retcode'] == salt.defaults.exitcodes.EX_OK
    and recursive and 'No unfinished transactions left.' not in cmd_ret_list[-1]['stdout']):
```
Or (probably better):
```python
if (cmd_ret_list[-1]['retcode'] == salt.defaults.exitcodes.EX_OK and recursive
    and 'No unfinished transactions left.' not in cmd_ret_list[-1]['stdout']):
```
But your miles may wary. Your call. :wink:

Reply:
@isbm I personally agree with "operator first", but I have seen the "operator last" in formatting examples and a bit in salt's own codebase (although only ~1/4 as often after a quick search in hindsight) so I went that route. Merged now, not sure why github didn't show a notification when I checked earlier.